File: 2486-most-frequent-even-element/most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        nums.sort()
        freq = {}
        for num in nums:
            if num % 2 == 0:
                freq[num] = freq.get(num, 0) + 1
        
        most_frequent = -1
        max_count = 0
        for key, count in freq.items():
            if count > max_count or (count == max_count and key < most_frequent):
                most_frequent = key
                max_count = count
                
        return most_frequent

File: 2486-most-frequent-even-element/most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer, Integer> fre = new HashMap<>();
        for(int num: nums){
            if(num%2 == 0)
                fre.put(num, fre.getOrDefault(num, 0)+1);
        }
        int maxCount = 0;
        int mostFrequency = -1;
        for(int key: fre.keySet()){
            int count = fre.get(key);
            if(maxCount < count || (maxCount == count && mostFrequency > key)){
                mostFrequency = key;
                maxCount = count;
            }
        }
        return mostFrequency;
    }
}